settingengine.go

```diff
@@ -114,3 +116,28 @@ func (e *SettingEngine) SetNetworkTypes(candidateTypes []NetworkType) {
 func (e *SettingEngine) SetInterfaceFilter(filter func(string) bool) {
 	e.candidates.InterfaceFilter = filter
 }
+
+// SetNAT1To1IPs has a list of external IP addresses of 1:1 (D)NAT.
+// This is useful when you are host a server using Pion on an AWS EC2 instance
+// which has a private address, behind a 1:1 DNAT with a public IP (e.g.
+// Elastic IP). In this case, you can give the public IP address so that
+// Pion will use the public IP address in its candidate instead of the private IP
+// address.
+func (e *SettingEngine) SetNAT1To1IPs(ips []string) {
+	e.candidates.NAT1To1IPs = ips
+}
+
+// SetNAT1To1IPCandidate is used along with SetNAT1To1IPs, to tell Pion which
+// type of candidate should use the given public IP address.
+// Two types of candidates are supported:
+// - "host": The public IP address will be used for the host candidate in the SDP.
+// - "srflx": A server reflexive candidate with the given public IP address will be added
+// to the SDP. If you choose "host", then the private IP address won't be advertised with
+// the peer. Also, this option cannot be used along with mDNS.
+// If you choose "srflx", it simply adds a server reflexive candidate with the public IP.
+// The host candidate is still available along with mDNS capabilities unaffected.
+// Please note that you cannot give STUN server URL at the same time. It will result in
+// an error otherwise.
+func (e *SettingEngine) SetNAT1To1IPCandidate(candidate string) {
```

Comment:
I would use `ICECandidateType` and then assert that only `ICECandidateTypeSrflx` and `ICECandidateTypeHost` are accepted

Reply:
I had thought for JavaScript support (via WASM), use of string was better. But ICECandidateTypeXxx has already been used, I will switch to it as suggested!
Thanks!

Reply:
Fixed at 70babf2